src/components/booking/BookingForm.tsx

```diff
@@ -0,0 +1,142 @@
+// // BookingForm.tsx
+import { Box, Divider, Stack, Typography } from '@mui/material';
+import { Dayjs } from 'dayjs';
+import React, { useCallback, useEffect, useState } from 'react';
+
+import { DatePicker } from '@mui/x-date-pickers/DatePicker';
+import { useGetAvailableDates } from '@src/api/queries/booking/useGetAvailableDates';
+import { styles } from '@src/pages/accomodation/Accommodation.styles';
+import ButtonPrimary from '../../components/button/ButtonPrimary';
+import {
+  DateFormat,
+  ReservationData,
+  createReservationData,
+  selectDatesType,
+  selectedDateType,
+} from './time';
+
+interface BookingFormProps {
+  onSubmit: (reservationData: {
+    startDate: string;
+    endDate: string;
+    accommodationId: string;
+  }) => void;
+  accomodationId: unknown;
+  price: number;
+}
+
+const BookingForm: React.FC<BookingFormProps> = ({ onSubmit, accomodationId, price }) => {
+  const { data, isPending, isError } = useGetAvailableDates(accomodationId as string);
+
+  const [selectedDates, setSelectedDates] = useState<selectDatesType>([null, null]);
+  const [startDate, endDate] = selectedDates;
+  const { availableDates, accommodationId } = data || {};
+  const [errorMessage, setErrorMessage] = useState('');
+  const [totalPrice, setTotalPrice] = useState<number>();
+
+  const shouldDisableDate = (date: Dayjs) => {
+    const dateString = date.format(DateFormat);
+    // Disable dates that are not within the available range
+    return !availableDates?.some(([start, end]) => dateString >= start && dateString <= end);
+  };
+
+  const handleDateChange = (index: number) => (newValue: selectedDateType) => {
+    setSelectedDates((prevSelectedDates) => {
+      const newSelectedDates = [...prevSelectedDates];
+      newSelectedDates[index] = newValue;
+      const [startDate, endDate] = newSelectedDates;
+      if (
+        (startDate && endDate && startDate?.isSame(endDate)) ||
+        (startDate && endDate && startDate.isAfter(endDate))
+      ) {
+        setErrorMessage('Check-in date must be before the check-out date and must not be same');
+      } else {
+        setErrorMessage('');
+      }
+
+      return newSelectedDates as selectDatesType;
+    });
+  };
+
+  const calculateTotalPrice = useCallback(() => {
+    if (startDate && endDate) {
+      const nights = endDate.diff(startDate, 'days');
+      const total = nights * price;
+      setTotalPrice(total);
+    }
+  }, [startDate, endDate, price]);
+
+  useEffect(() => {
+    if (!errorMessage) {
+      calculateTotalPrice();
+    }
+  }, [calculateTotalPrice, startDate, endDate, errorMessage]);
+
+  const handleReserveButtonClick = useCallback(
```

Comment:
if the user selects the same date, the button is not disabled, it sends a requests